hie-plugin-api/Haskell/Ide/Engine/Cradle.hs

```diff
@@ -14,8 +33,587 @@ import HIE.Bios as BIOS
 -- of the project that may or may not be accurate.
 findLocalCradle :: FilePath -> IO Cradle
 findLocalCradle fp = do
-  -- Get the cabal directory from the cradle
   cradleConf <- BIOS.findCradle fp
   case cradleConf of
-    Just yaml -> BIOS.loadCradle yaml
-    Nothing -> BIOS.loadImplicitCradle fp
\ No newline at end of file
+    Just yaml -> fixCradle <$> BIOS.loadCradle yaml
+    Nothing   -> cabalHelperCradle fp
+
+-- | Check if the given cradle is a stack cradle.
+-- This might be used to determine the GHC version to use on the project.
+-- If it is a stack-cradle, we have to use `stack path --compiler-exe`
+-- otherwise we may ask `ghc` directly what version it is.
+isStackCradle :: Cradle -> Bool
+isStackCradle = (`elem` ["stack", "Cabal-Helper-Stack", "Cabal-Helper-Stack-None"])
+  . BIOS.actionName
+  . BIOS.cradleOptsProg
```

Comment:
We can not make `actionName` a sum type, because then we can not add a cradle that hie-bios doesnt know about, like in this case "Cabal-Helper-Stack", except by adding a custom constructor such as `Other String`, which in my opinion defeats the purpose of it.
I talked to @mpickering to add predicates upstream that hide this string check.

Reply:
The reason this check is to hack around two things.

1. The options that the "stack" cradle from hie-bios doesn't really return the correct options. So there is a hack to add an additional target. 
2. The cradle doesn't specify the version of GHC which is expected to be used to compile a project. For a cabal project we could just configure the project with the correct version of GHC (the one which was used to build that specific version of HIE) but that wouldn't work for stack projects which are locked to a specific compiler version.

So if we fix both of these things we can also remove this check. That seems like a better long term solution.